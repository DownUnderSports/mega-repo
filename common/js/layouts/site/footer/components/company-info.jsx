import React, { PureComponent } from 'react';
import CopyClip from 'common/js/helpers/copy-clip'
/*
<br/>
Physical Location: <br className='d-sm-none'/><a
  data-content='physical'
  href="https://maps.google.com/?q=Down Under Sports, 1755 N 400 E #201, North Logan, UT 84341-6010">
  1755 N 400 E Ste 201, North Logan, UT 84341-6010
</a>
*/
export default class CompanyInfo extends PureComponent {
  copyMailingAddress(e) {
    e.preventDefault()
    CopyClip.prompted(e.target.innerText)
  }

  render() {
    return (
      <address>
        <strong>
          <i>
            Down Under Sports
          </i>
        </strong>
        <br/>
        Phone (Call/Text): <a data-content='phone' href="tel:+1-435-753-4732">435-753-4732</a>
        <br/>
        Email: <a data-content='email' href="mailto:mail@downundersports.com">mail@downundersports.com</a>
        <br/>
        Mailing Address: <br className='d-sm-none'/><a
          data-content='mailing'
          href="/"
          onClick={this.copyMailingAddress}
        >
          PO Box 6010, North Logan, UT 84341-6010
        </a>
      </address>
    )
  }
}
