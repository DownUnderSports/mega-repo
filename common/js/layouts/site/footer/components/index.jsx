export { default as FooterLinks } from './footer-links'
export { default as CompanyInfo } from './company-info'
