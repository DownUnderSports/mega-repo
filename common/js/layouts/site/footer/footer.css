.Site-footer {
  position: relative;
  background-color: #222;
  color: #ccc;
}
@media screen and (max-width: 310px) {
  .Site-footer {
    display: none;
  }
}
@media print {
  .Site-footer {
    display: none;
  }
}
.Site-footer address {
  margin: 0;
  font-size: 1rem;
}
@media screen and (min-width: 768px) {
  .Site-footer address {
    white-space: nowrap;
  }
}
.Site-footer address a {
  color: #61dafb;
}
.Site-footer table {
  font-size: 1rem;
  white-space: nowrap;
  overflow: scroll;
}
.Site-footer table.table {
  width: unset;
  margin: unset;
}
.Site-footer table.table td,
.Site-footer table.table th {
  padding: 0.25rem;
}
.Site-footer table.table thead th,
.Site-footer table.table thead td {
  border-top: none;
  font-size: 1.15rem;
}
.Site-footer table.table tbody th {
  padding-right: 4rem;
}
@media screen and (max-width: 767px) {
  .Site-footer table.table tbody th {
    padding: 0.25rem;
  }
}
.Site-footer .footer-content {
  padding: 20px;
}
.Site-footer .footer-content,
.Site-footer .footer-content .footer-info-wrapper,
.Site-footer .footer-content .footer-office-hours {
  display: flex;
  flex-flow: column;
}
.Site-footer .footer-content .footer-office-hours {
  margin-bottom: 0.5rem;
}
@media screen and (max-width: 369px) {
  .Site-footer .footer-content .footer-office-hours table.table tbody tr {
    display: flex;
    flex-flow: column nowrap;
  }
  .Site-footer .footer-content .footer-office-hours table.table tbody tr td,
  .Site-footer .footer-content .footer-office-hours table.table tbody tr th {
    padding: 0.25rem;
  }
  .Site-footer .footer-content .footer-office-hours table.table tbody tr td {
    padding-top: 0;
    border: 0;
  }
}
@media screen and (min-width: 992px) {
  .Site-footer .footer-content .footer-info-wrapper,
  .Site-footer .footer-content .footer-office-hours {
    flex-flow: row;
  }
  .Site-footer .footer-content .footer-office-hours {
    margin-bottom: 0.5rem;
    justify-content: flex-end;
  }
}
.Site-footer .footer-content .footer-link {
  text-decoration: none;
  text-decoration: none;
  display: block;
  white-space: nowrap;
  transform: scale(0.97);
  transition: transform 0.1s ease-in-out;
  height: 45px;
  font-size: 1rem;
  margin: 1rem 0;
  border: 1px solid #61dafb;
  background-color: rgba(0,0,0,0);
  color: #61dafb;
}
.Site-footer .footer-content .footer-link:hover {
  background-color: #61dafb;
  color: #000;
}
@media screen and (min-width: 1024px) {
  .Site-footer .footer-content .footer-link {
    font-size: 1.15rem;
  }
}
@media screen and (min-width: 1536px) {
  .Site-footer .footer-content .footer-link {
    margin: unset;
    flex: 1;
  }
}
@media screen and (min-width: 1920px) {
  .Site-footer .footer-content .footer-link {
    height: 60px;
    font-size: 1.25rem;
  }
}
.Site-footer .footer-content .footer-link:hover {
  transform: scale(1);
}
.Site-footer .footer-content .footer-office-hours,
.Site-footer .footer-content .footer-menu {
  padding-top: 0.5rem;
  margin-top: 0.5rem;
}
.Site-footer .footer-content .footer-menu {
  position: relative;
  font-size: 0;
  width: 100%;
  max-width: 100vw;
  overflow: auto;
  border-top: 1px solid #ccc;
}
@media screen and (min-width: 1536px) {
  .Site-footer .footer-content .footer-menu {
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
    justify-content: space-between;
  }
}
@media screen and (min-width: 1920px) {
  .Site-footer .footer-content .footer-office-hours {
    padding-top: unset;
    margin-top: unset;
  }
}
