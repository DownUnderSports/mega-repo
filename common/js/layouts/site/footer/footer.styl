.Site-footer
  position: relative
  background-color: #222
  color: #ccc
  @media screen and (max-width: 310px)
    display: none
  @media print
    display: none

  address
    margin: 0
    font-size: 1rem
    @media screen and (min-width: 768px)
      white-space: nowrap
    a
      color: rgba(97,218,251,1)

  table
    font-size: 1rem
    white-space: nowrap
    overflow: scroll
    &.table
      width: unset
      margin: unset
      td
      th
        padding: .25rem
      thead
        th
        td
          border-top: none
          font-size: 1.15rem
      tbody
        th
          padding-right: 4rem
          @media screen and (max-width: 767px)
            padding: .25rem

  .footer-content
    padding: 20px
    &
    .footer-info-wrapper
    .footer-office-hours
      display: flex
      flex-flow: column
    .footer-office-hours
      margin-bottom: .5rem

      @media screen and (max-width: 369px)
        table.table
          tbody
            tr
              display: flex
              flex-flow: column nowrap
              td
              th
                padding: .25rem
              td
                padding-top: 0
                border: 0


    @media screen and (min-width: 992px)
      .footer-info-wrapper
      .footer-office-hours
        flex-flow: row
      .footer-office-hours
        margin-bottom: .5rem
        justify-content: flex-end

    .footer-link
      text-decoration: none
      text-decoration: none
      display: block
      white-space: nowrap
      transform: scale(0.97)
      transition: transform 0.1s ease-in-out
      height: 45px
      font-size: 1rem
      margin: 1rem 0
      border: 1px solid rgba(97,218,251,1)
      background-color: rgba(0,0,0,0)
      color: rgba(97,218,251,1)

      &:hover
        background-color: rgba(97,218,251,1)
        color: rgb(0,0,0)

      // @media screen and (min-width: 360px)
      //   &.bbb-logo
      //     height: 45px
      //     width: 30px

      @media screen and (min-width: 1024px)
        font-size: 1.15rem

      @media screen and (min-width: 1220px)
        font-size: 14px
        margin: unset
        flex: 1

      @media screen and (min-width: 1536px)
        font-size: 16px

      @media screen and (min-width: 1920px)
        height: 60px
        font-size: 20px

      &:hover
        transform: scale(1)
        // &.bbb-logo
        //   background-position: 100% 0%

    .footer-office-hours
    .footer-menu
      padding-top: .5rem
      margin-top: .5rem

    .footer-menu
      position: relative
      font-size: 0
      width: 100%
      max-width: 100vw
      overflow: auto
      // @media screen and (min-width: 992px)
      border-top: 1px solid #ccc

      @media screen and (min-width: 1220px)
        display: flex
        flex-flow: row nowrap
        align-items: center
        justify-content: space-between

    @media screen and (min-width: 1920px)
      .footer-office-hours
        padding-top: unset
        margin-top: unset
