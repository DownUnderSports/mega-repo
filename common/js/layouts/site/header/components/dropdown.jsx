import React, { Component } from 'react';
import {arrayOf, oneOfType, shape, string, node} from 'prop-types';
import HeaderLink from './header-link'

const linkShape = shape({
  to: string,
  children: node
})

export default class Dropdown extends Component {
  static propTypes = {
    links: arrayOf(linkShape).isRequired,
    header: oneOfType([
      string,
      node
    ]).isRequired
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.closeClick)
  }

  state = { click: false, hover: false }

  toggleClick = () => {
    console.log('click', this.state.click)
    this.setState({ click: !this.state.click, hover: false }, this.setClickListener)
  }
  closeClick = () => {
    console.log('close click', this.state.click)
    document.removeEventListener('click', this.closeClick)

    this.setState({ click: false, hover: false })
  }
  openHover = () => this.setState({ hover: true })
  closeHover = () => this.setState({ hover: false })

  setClickListener = () => {
    document.removeEventListener('click', this.closeClick)
    this.state.click && document.addEventListener('click', this.closeClick)
  }

  render() {
    return (
      <div
        className={`dropdown clickable nav-link ${this.state.click || this.state.hover}`}
        onMouseEnter={this.openHover}
        onMouseLeave={this.closeHover}
      >
        <div className="dropdown-title" onClick={this.toggleClick}>
          { this.props.header }
          <i className="material-icons">arrow_drop_down</i>
        </div>
        <ul className="dropdown-list">
          {
            this.props.links.map(
              (link, i) =>
                <li key={i} className="dropdown-link">
                  {<HeaderLink key={i} {...link} />}
                </li>
            )
          }
        </ul>
      </div>
    )
  }
}
