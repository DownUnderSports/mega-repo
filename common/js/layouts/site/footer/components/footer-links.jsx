import React, { PureComponent } from 'react';
import {arrayOf, shape, string, node} from 'prop-types';

import { Link } from 'react-component-templates/components'

export default class FooterLinks extends PureComponent {
  static propTypes = {
    links: arrayOf(shape({
      to: string,
      children: node
    })).isRequired
  }

  mapLink = (props = {}, i) =>
    <Link
      key={i}
      className="footer-link d-flex justify-content-center align-items-center px-3 font-weight-bolder"
      {...props}
    />

  render() {
    const { links = [] } = this.props;
    return (<div className="footer-menu">
      { links.map(this.mapLink) }
    </div>)
  }
}
