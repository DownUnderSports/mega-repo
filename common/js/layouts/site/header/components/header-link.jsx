import React, { PureComponent } from 'react';
import { Link } from 'react-component-templates/components'

export default class HeaderLink extends PureComponent {
  render() {
    const { className, ...props } = this.props
    return (
      <Link
        className={`nav-link header-link ${className || ''}`}
        {...props}
      />
    )
  }
}
