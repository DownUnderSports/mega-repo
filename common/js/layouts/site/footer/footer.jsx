import React, { PureComponent } from 'react'

import { CompanyInfo, FooterLinks } from './components'
import './footer.css';
import insurancePDF from 'common/assets/pdfs/insurance.pdf'
// import fb from 'common/assets/images/facebook/blue.svg'
// import travelex from 'common/assets/images/travelex.png'
// import bbbFront from 'common/assets/images/bbb/bbb-front.png'
// import bbbBack from 'common/assets/images/bbb/bbb-back.png'

// const dateToLocalTime = (d) => d.toLocaleTimeString('en-us',{timeZoneName:'short'}).replace(/:0+/g, '').replace(/[A-Z]+$/, '($&)')

/* MDT LIMITED */
// const startTime    = new Date('2019-08-31T20:00:00Z').toLocaleTimeString().replace(/:0+/g, ''),
//       endTime  = dateToLocalTime(new Date('2019-08-31T22:00:00Z'))

export default class Footer extends PureComponent {
  links = [
    {
      to: 'https://www.facebook.com/DownUnderSports',
      target: '_blank',
      rel: 'noopener noreferrer',
      children: [ 'Facebook' ],
    },
    {
      to: "https://www.bbb.org/us/ut/north-logan/profile/athletic-organizations/down-under-sports-1166-2001870/#sealclick",
      target: '_blank',
      rel: 'noopener noreferrer',
      children: [ 'Better Business Bureau' ],
    },
    {
      to: insurancePDF,
      target: '_insurance',
      rel: 'noopener noreferrer',
      children: [ 'Travelex Insurance' ],
    },
    {
      to: '/travel-giveaways',
      children: [ 'Travel Giveaway Rules' ],
    },
    {
      to: '/open-tryouts',
      children: [ 'Open Tryouts' ],
    },
    {
      to: "/privacy-policy",
      children: [ 'Privacy Policy' ],
    },
    {
      to: '/refunds',
      children: [ 'Refund Policy' ],
    },
    {
      to: '/terms',
      children: [ 'Program Terms' ],
    },
  ]

  render(){
    return (
      <footer className={this.props.className || 'Site-footer'}>
        <div className="footer-content container-fluid">
          <div className="footer-info-wrapper w-100">
            <CompanyInfo />
            <div className="footer-office-hours w-100">
              <table className="table text-light">
                <thead>
                  <tr>
                    <th colSpan="2">
                      Office Hours (Limited)
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <th>
                      Tuesday - Thursday
                    </th>
                    <td>
                      {/*
                      {startTime} - {endTime}
                      */}
                      Varied
                    </td>
                  </tr>
                  <tr>
                    <th>
                      Friday - Monday
                    </th>
                    <td>
                      Closed
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <FooterLinks
            links={this.links}
          />
        </div>
      </footer>
    )
  }
}
