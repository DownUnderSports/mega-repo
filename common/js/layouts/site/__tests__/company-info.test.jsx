import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import { CompanyInfo } from '../footer/components'

describe('Layouts::components - Footer::CompanyInfo', () => {
  const div = document.createElement('div');

  const createCompanyInfo = ({...props}) => {
    ReactDOM.render((
      <Router>
        <CompanyInfo {...props} />
      </Router>
    ), div);
    return div.querySelector('address')
  }

  it('renders a sematic address', () => {
    const rendered = createCompanyInfo()
    expect(rendered).toBeTruthy()
    expect(rendered.tagName).toBe("ADDRESS")
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the company name', () => {
    const name = createCompanyInfo().querySelector('strong i')
    expect(name).toBeTruthy()
    expect(/down under sports/i.test(name.textContent)).toBeTruthy()
    ReactDOM.unmountComponentAtNode(div);
  })

  it('contains phone number, email, mailing address, and physical address', () => {
    const rendered = createCompanyInfo()

    const email = rendered.querySelector('a[data-content=email]'),
          phone = rendered.querySelector('a[data-content=phone]'),
          mailing = rendered.querySelector('a[data-content=mailing]'),
          physical = rendered.querySelector('a[data-content=physical]');

    expect(email).toBeTruthy()
    expect(/^mailto:/.test(email.href)).toBeTruthy()
    expect(email.tagName).toBe("A")

    expect(phone).toBeTruthy()
    expect(/^tel:/.test(phone.href)).toBeTruthy()
    expect(phone.tagName).toBe("A")

    expect(mailing).toBeTruthy()
    expect(/^PO Box/.test(mailing.innerHTML)).toBeTruthy()
    expect(mailing.tagName).toBe("A")

    expect(physical).toBeTruthy()
    expect(/^1755 N/.test(physical.innerHTML)).toBeTruthy()
    expect(physical.tagName).toBe("A")

    ReactDOM.unmountComponentAtNode(div);
  })

  it('is snapshotable', () => {
    const tree = renderer
      .create(
        <Router>
          <CompanyInfo />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
  })
})
