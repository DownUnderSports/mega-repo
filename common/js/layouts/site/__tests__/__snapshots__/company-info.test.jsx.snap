// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layouts::components - Footer::CompanyInfo is snapshotable 1`] = `
<address>
  <strong>
    <i>
      Down Under Sports
    </i>
  </strong>
  <br />
  Phone: 
  <a
    data-content="phone"
    href="tel:+1-435-753-4732"
  >
    435-753-4732
  </a>
  <br />
  Email: 
  <a
    data-content="email"
    href="mailto:mail@downundersports.com"
  >
    mail@downundersports.com
  </a>
  <br />
  Mailing Address: 
  <br
    className="d-sm-none"
  />
  <a
    data-content="mailing"
    href="/"
    onClick={[Function]}
  >
    PO Box 6010, North Logan, UT 84341-6010
  </a>
  <br />
  Physical Location: 
  <br
    className="d-sm-none"
  />
  <a
    data-content="physical"
    href="https://maps.google.com/?q=Down Under Sports, 1755 N 400 E, Ste 201, North Logan, UT 84341-6010"
  >
    1755 N 400 E, Ste 201, North Logan, UT 84341-6010
  </a>
</address>
`;
